src/Symfony/Component/Ldap/Adapter/AbstractQuery.php

```diff
@@ -32,12 +32,12 @@ public function __construct(ConnectionInterface $connection, string $dn, string
             'maxItems' => 0,
             'sizeLimit' => 0,
             'timeout' => 0,
-            'deref' => static::DEREF_NEVER,
+            'deref' => QueryInterface::DEREF_NEVER,
```

Comment:
This will break backwards compatibility, as you could use them in a child implementation and overwrite them. @csarrazi do you know if that was intentional?

Reply:
This is, but it's a design issue too.

Reply:
@iltar I can not make it deprecated and fixed it in the same time. But i dont think this is ever used.

Reply:
You can trigger the deprecation, which should cause developers to fix the issue. It'll have to be fixed in 5.0. It's up to the core members to decide whether or not this is an acceptable BC break

Reply:
Actually, one of the things I'm unsatisfied with is the fact that the values in QueryInterface are actually extracted from the `ext_ldap` implementation, which is actually problematic as this means that the implementation leaks into the interface.

This was actually the rationale behind using `static::` instead of QueryInterface, as the goal was that other implementations could use other values.

For 5.x, the actual values used in the QueryInterface should instead be defined in the actual implementation instead of the interface, and the interface constants should rather be string values.

Reply:
I agree. We should create a `LdapOption`, which would contain string values for each constant.
The actual value (used by `ext_ldap` should be limited to the actual implementation, so should only be present in the query implementation.